### PR TITLE
Remove Infinite Loop

### PR DESCRIPTION
**Pull request by Maithem**
## Overview
Removed an unnecessary loop from the MultiCheckpointWriter.

Why should this be merged: Lower layers have bounded retries, infinitely loop in the MultiCheckpointWriter is not good.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

**Comment by StrikingThirteen** [comment]
It seems that with removal of infinite loop, we only attempt once to append the checkpoint. Wouldn't it be possible to go for a middle ground logic? For example attempting for a limited number times (e.g. 3) after which we won't attempt any more. As a result there won't be any infinite attempts yet we have more resilience in case of a recoverable TransactionAbortException.

File: runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java [hunk 1]
```diff
@@ -80,28 +80,27 @@ public long appendCheckpoints(CorfuRuntime rt, String author,
...
+                cpw.setPostAppendFunc(postAppendFunc);
+                log.trace("appendCheckpoints: checkpoint map {} begin",
+                        Utils.toReadableId(map.getCorfuStreamID()));
+                try {
+                    List<Long> addresses = cpw.appendCheckpoint();
+                    log.trace("appendCheckpoints: checkpoint map {} end",
+                            Utils.toReadableId(map.getCorfuStreamID()));
+                    checkpointLogAddresses.addAll(addresses);
+                } catch (TransactionAbortedException ae) {
+                    log.warn("appendCheckpoints: checkpoint map {} "
```
[the remark this answers is not in the document]
Comment (Maithem) [reply runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java]: done.

**Comment by Maithem** [comment]
@StrikingThirteen the thread calling this should have a retry. Also, lower levels do have bounded retries.

**Comment by corfu-repo** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [MultiCheckpointWriter.java#L15](https://github.com/CorfuDB/CorfuDB/blob/e5b9e6a72cddb4fce244a35f568aa4291e3e5b5d/runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java#L15): Remove this unused import 'org.corfudb.runtime.exceptions.TransactionAbortedException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by StrikingThirteen** [comment]
@Maithem Sounds good. LGTM. 
One question: where in the lower levels we have a bounded retries? It seems that this leads to CheckpointWriter's appendCheckpoint() which doesn't specifically have bounded retries (at that level).

**Comment by Maithem** [comment]
@StrikingThirteen eventually, it will hit the stream layer. The stream layer append has bounded retries on appends.

https://github.com/CorfuDB/CorfuDB/blob/master/runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java#L77

**Comment by Maithem** [comment]
@zalokhan the checkpoint writer doesn't do transactional writes, so I'm not sure what you are referring to. Please explain.

**Comment by corfu-repo** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [MultiCheckpointWriter.java#L15](https://github.com/CorfuDB/CorfuDB/blob/0a3b6e2890c7b39e06927e70e2192a69e1b23423/runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java#L15): Remove this unused import 'org.corfudb.runtime.exceptions.TransactionAbortedException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by StrikingThirteen** [comment]
@Maithem isn't this what @zalokhan highlighted?
https://github.com/CorfuDB/CorfuDB/blob/7c7d2e89c2a820f5928e6ea75bde4dd11e26e4bb/runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java#L152

**Comment by zalokhan** [comment]
@Maithem You are right, it is a snapshot transaction. I got confused.
In the current issue, we see a ShutdownException wrapped in a RuntimeException wrapped in a TransactionAbortedException thrown by the TxEnd.

**Comment by corfu-repo** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [MultiCheckpointWriter.java#L15](https://github.com/CorfuDB/CorfuDB/blob/42a33dadd4072883be32f6251c7d3d6f09b5ba1f/runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java#L15): Remove this unused import 'org.corfudb.runtime.exceptions.TransactionAbortedException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Maithem** [comment]
@StrikingThirteen  @zalokhan You're right about it being an TransactionAbortedException, but I was saying that it can be ignored since its a read only transaction.

**Comment by corfu-repo** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [MultiCheckpointWriter.java#L15](https://github.com/CorfuDB/CorfuDB/blob/b4bacdf81b64f8d2bca229cee3749c82865876ac/runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java#L15): Remove this unused import 'org.corfudb.runtime.exceptions.TransactionAbortedException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**